daemon/app/Main.hs

```diff
@@ -165,9 +165,15 @@ webServer var = do
     \_ -> loopM step (emptyUIState initTime, ss0)
   where
     step :: (UIState, ServerState) -> Widget IHTML (Either (UIState, ServerState) ())
-    step (ui, ss) = do
-      let lastUpdatedUI = ui ^. uiLastUpdated
-          lastUpdatedServer = ss ^. serverLastUpdated
+    step (ui0, ss) = do
+      let lastUpdatedUI = ui0 ^. uiLastUpdated
+      stepStartTime <- unsafeBlockingIO getCurrentTime
```

Comment:
It makes sense that the time stamp is always taken at the beginning of the step and mark whether UI should be updated or not right away.